### ISSUE TITLE
FullTextSearch stopped working and won't reinstall successfully

### ISSUE THREAD

**Issue by BetaLeaf** [nextcloud/vm#1129]
## Steps to reproduce
1. Attempt to reinstall FullTextSearch.sh ("sudo bash /var/scripts/apps/fulltextsearch/sh")

## Expected behaviour
It should work

## Actual behaviour
Doesn't work correctly

## Server configuration

**Nextcloud server version:** (see your admin page)
18.0.1

**Server OS** (Ubuntu server is default)
Ubuntu 18.04.3

**How did you install the VM?** (Scripted install from master **OR** Released version)
Scripted Install from Master

## Network
**Do you use DHCP?**
Yes, but ip assigned on router

**Is port 80 and/or 443 open?**
80 is open but not being used by Nextcloud (cameyo server on port 80)
443 is open and is the intended port to use

#### Installation log (command output)
```
ncadmin@cloud:~$ sudo bash /var/scripts/apps/fulltextsearch.sh
[sudo] password for ncadmin:
Installing Elastic Search & Full Text Search on Nextcloud...
Removing old version of Full Text Search and resetting the app...

In RunningService.php line 86:

  Index is already running


fulltextsearch:reset [--output [OUTPUT]] [--] [<provider>]

fulltextsearch disabled
fulltextsearch_elasticsearch disabled
files_fulltextsearch disabled
vm.max_map_count=262144
Max map count already set, skipping...
Press any key to continue. Press CTRL+C to abort
ffcad70a2e1a
Deleted Containers:
ffcad70a2e1affd84678c83af8301027d9851f883cfa80eb2c52a5ea1870d549

Total reclaimed space: 464.9kB
Deleted Images:
untagged: ark74/nc_fts:latest
untagged: ark74/nc_fts@sha256:ad184799b34904d44f9f1f26fc03f20c74c543de2181cd02a45af31db7e33377
deleted: sha256:4e6f05f7a75617aae94b26332559842db61c4b43184584f8da4ad2e4fb705580
deleted: sha256:f4506f3b910ac7bf88a51222db67de3804c0dcfe538d60866f9cb14c2de49dc2
deleted: sha256:099ff5b4bc19313d1ece09920d705d367e9a25bce1aa5fa1f916a79798f1a9ad
deleted: sha256:b928a21b4b8a5000e94ee877b74b8f9424e7f6c1a7227c2c7f64c7bd83417681
deleted: sha256:f1f56385da7344dec2012547725112f921473d1e37ce9a783e72592af744d5a8
deleted: sha256:0bb6f13cb5a1d01e57d78466f930fc600bbdf19b66275bdd8e92d8200f577964
deleted: sha256:fe96c21169db877ebea5ef8f47d2ca6af0da16cbd05fb67ec84e0bb0e02310b4
deleted: sha256:eb935d0cb5f239eedad9377f201ac7574438e1e94c07c366e92df11c66580a21
deleted: sha256:77b174a6a187b610e4699546bd973a8d1e77663796e3724318a2a4b24cb07ea0

Total reclaimed space: 780.7MB
Deleted Volumes:
esdata

Total reclaimed space: 20.6kB
Unable to find image 'ark74/nc_fts:latest' locally
latest: Pulling from ark74/nc_fts
ab5ef0e58194: Pull complete                                                                                                                                             8a5003fa724c: Pull complete                                                                                                                                             11141d076ab6: Pull complete                                                                                                                                             1afda89faaf5: Pull complete                                                                                                                                             229ceda2870e: Pull complete                                                                                                                                             30e0992fc2d7: Pull complete                                                                                                                                             3e3d7f2d6890: Pull complete                                                                                                                                             a6730b797ed0: Pull complete                                                                                                                                             Digest: sha256:ad184799b34904d44f9f1f26fc03f20c74c543de2181cd02a45af31db7e33377
Status: Downloaded newer image for ark74/nc_fts:latest
266d35f71f3cbc9fcd6538be25792fbb692d52958c1baf8123ea448c8ede058c
fts_esror
Waiting for docker bootstraping...
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
{"type": "server", "timestamp": "2020-03-01T04:40:28,949Z", "level": "INFO", "component": "o.e.e.NodeEnvironment", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/sda2)]], net usable_space [75.9gb], net total_space [124gb], types [ext4]" }
{"type": "server", "timestamp": "2020-03-01T04:40:28,954Z", "level": "INFO", "component": "o.e.e.NodeEnvironment", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "heap size [494.9mb], compressed ordinary object pointers [true]" }
{"type": "server", "timestamp": "2020-03-01T04:40:28,958Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "node name [266d35f71f3c], node ID [TdP_JH2kQBa4scELYkMlRQ], cluster name [docker-cluster]" }
{"type": "server", "timestamp": "2020-03-01T04:40:28,959Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "version[7.5.2], pid[1], build[oss/docker/8bec50e1e0ad29dad5653712cf3bb580cd1afcdf/2020-01-15T12:11:52.313576Z], OS[Linux/4.15.0-88-generic/amd64], JVM[AdoptOpenJDK/OpenJDK 64-Bit Server VM/13.0.1/13.0.1+9]" }
{"type": "server", "timestamp": "2020-03-01T04:40:28,960Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "JVM home [/usr/share/elasticsearch/jdk]" }
{"type": "server", "timestamp": "2020-03-01T04:40:28,960Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=COMPAT, -Xms1g, -Xmx1g, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -Djava.io.tmpdir=/tmp/elasticsearch-7847208057111874150, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=data, -XX:ErrorFile=logs/hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Des.cgroups.hierarchy.override=/, -Xms512M, -Xmx512M, -XX:MaxDirectMemorySize=268435456, -Des.path.home=/usr/share/elasticsearch, -Des.path.conf=/usr/share/elasticsearch/config, -Des.distribution.flavor=oss, -Des.distribution.type=docker, -Des.bundled_jdk=true]" }
OpenJDK 64-Bit Server VM warning: Option UseConcMarkSweepGC was deprecated in version 9.0 and will likely be removed in a future release.
{"type": "server", "timestamp": "2020-03-01T04:40:42,723Z", "level": "INFO", "component": "o.e.e.NodeEnvironment", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "using [1] data paths, mounts [[/usr/share/elasticsearch/data (/dev/sda2)]], net usable_space [75.9gb], net total_space [124gb], types [ext4]" }
{"type": "server", "timestamp": "2020-03-01T04:40:42,735Z", "level": "INFO", "component": "o.e.e.NodeEnvironment", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "heap size [494.9mb], compressed ordinary object pointers [true]" }
{"type": "server", "timestamp": "2020-03-01T04:40:42,746Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "node name [266d35f71f3c], node ID [TdP_JH2kQBa4scELYkMlRQ], cluster name [docker-cluster]" }
{"type": "server", "timestamp": "2020-03-01T04:40:42,747Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "version[7.5.2], pid[1], build[oss/docker/8bec50e1e0ad29dad5653712cf3bb580cd1afcdf/2020-01-15T12:11:52.313576Z], OS[Linux/4.15.0-88-generic/amd64], JVM[AdoptOpenJDK/OpenJDK 64-Bit Server VM/13.0.1/13.0.1+9]" }
{"type": "server", "timestamp": "2020-03-01T04:40:42,747Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "JVM home [/usr/share/elasticsearch/jdk]" }
{"type": "server", "timestamp": "2020-03-01T04:40:42,750Z", "level": "INFO", "component": "o.e.n.Node", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "JVM arguments [-Des.networkaddress.cache.ttl=60, -Des.networkaddress.cache.negative.ttl=10, -XX:+AlwaysPreTouch, -Xss1m, -Djava.awt.headless=true, -Dfile.encoding=UTF-8, -Djna.nosys=true, -XX:-OmitStackTraceInFastThrow, -Dio.netty.noUnsafe=true, -Dio.netty.noKeySetOptimization=true, -Dio.netty.recycler.maxCapacityPerThread=0, -Dio.netty.allocator.numDirectArenas=0, -Dlog4j.shutdownHookEnabled=false, -Dlog4j2.disable.jmx=true, -Djava.locale.providers=COMPAT, -Xms1g, -Xmx1g, -XX:+UseConcMarkSweepGC, -XX:CMSInitiatingOccupancyFraction=75, -XX:+UseCMSInitiatingOccupancyOnly, -Djava.io.tmpdir=/tmp/elasticsearch-11463854858861839092, -XX:+HeapDumpOnOutOfMemoryError, -XX:HeapDumpPath=data, -XX:ErrorFile=logs/hs_err_pid%p.log, -Xlog:gc*,gc+age=trace,safepoint:file=logs/gc.log:utctime,pid,tags:filecount=32,filesize=64m, -Des.cgroups.hierarchy.override=/, -Xms512M, -Xmx512M, -XX:MaxDirectMemorySize=268435456, -Des.path.home=/usr/share/elasticsearch, -Des.path.conf=/usr/share/elasticsearch/config, -Des.distribution.flavor=oss, -Des.distribution.type=docker, -Des.bundled_jdk=true]" }
{"type": "server", "timestamp": "2020-03-01T04:40:49,070Z", "level": "INFO", "component": "t.b.r.e.LogBuildInfoMessage$", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "Starting ReadonlyREST plugin v1.19.0 on ES v7.5.2" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,017Z", "level": "INFO", "component": "t.b.r.c.RorSsl$", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "Cannot find SSL configuration is elasticsearch.yml, trying: /usr/share/elasticsearch/config/readonlyrest.yml" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,451Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [aggs-matrix-stats]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,452Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [analysis-common]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,455Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [ingest-common]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,456Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [ingest-geoip]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,456Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [ingest-user-agent]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,456Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [lang-expression]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,462Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [lang-mustache]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,462Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [lang-painless]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,466Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [mapper-extras]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,470Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [parent-join]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,474Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [percolator]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,475Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [rank-eval]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,478Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [reindex]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,481Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [repository-url]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,482Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded module [transport-netty4]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,488Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded plugin [ingest-attachment]" }
{"type": "server", "timestamp": "2020-03-01T04:40:50,488Z", "level": "INFO", "component": "o.e.p.PluginsService", "cluster.name": "docker-cluster", "node.name": "266d35f71f3c", "message": "loaded plugin [readonlyrest]" }
Installing fulltextsearch...
fulltextsearch enabled
Installing fulltextsearch_elasticsearch...
fulltextsearch_elasticsearch enabled
Installing files_fulltextsearch...
files_fulltextsearch enabled
{
    "search_platform": "OCA\\FullTextSearch_ElasticSearch\\Platform\\ElasticSearchPlatform",
    "app_navigation": "0",
    "provider_indexed": "{\"deck\":\"1\"}",
    "cron_err_reset": "1582304702",
    "tick_ttl": "1800"
}
{
    "elastic_host": "http:\/\/edlxvwhccfdnxwoqwmpwzcdme:PWVcJW0lEU73SHzZMjbepvrIf@localhost:9200",
    "elastic_index": "edlxvwhccfdnxwoqwmpwzcdme-index",
    "fields_limit": "10000",
    "es_ver_below66": "0",
    "analyzer_tokenizer": "standard"
}
{
    "files_local": "1",
    "files_external": "1",
    "files_group_folders": "1",
    "files_encrypted": "0",
    "files_federated": "0",
    "files_size": "20",
    "files_pdf": "1",
    "files_office": "1",
    "files_image": "0",
    "files_audio": "0"
}

Options: []
Memory: 37 MB
┌─ Indexing  ────
│ Action: fillDocument
│ Provider: Files                Account: BetaLeaf
│ Document: 2273860
│ Info: text/plain
│ Title: AnyDesk/connection_trace.txt
│ Content size:
│ Chunk:      2/645
│ Progress:      0/1
└──
┌─ Results ────
│ Result:      3/3
│ Index: deck:3
│ Status: ok
│ Message: {"_index":"edlxvwhccfdnxwoqwmpwzcdme-index","_type":"standard","_id
│ ":"deck:3","_version":1,"result":"created","_shards":{"total":2,"successful":
│ 1,"failed":0},"_seq_no":2,"_primary_term":1}
└──
┌─ Errors ────
│ Error:      0/0
│ Index:
│ Exception:
│ Message:
│
│
└──
## x:first result ## c/v:prec/next result ## b:last result
## f:first error ## h/j:prec/next error ## d:delete error ## l:last error
## q:quit ## p:pause
PHP Fatal error:  Declaration of daita\MySmallPhpTools\Db\ExtendedQueryBuilder::exprLimitToDBFieldArray(string $field, array $values, bool $eq = true, bool $cs = true, string $alias = ''): OCP\DB\QueryBuilder\ICompositeExpression must be compatible with daita\MySmallPhpTools\IExtendedQueryBuilder::exprLimitToDBFieldArray(string $field, array $values, bool $cs = true, string $alias = ''): OCP\DB\QueryBuilder\ICompositeExpression in /var/www/nextcloud/apps/social/vendor/daita/my-small-php-tools/lib/Db/ExtendedQueryBuilder.php on line 49
Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:index failed
```

**Comment by enoch85** [comment]
@szaimen @Ark74 Any thoughts?

**Comment by szaimen** [comment]
I just wonder why the path to the script is /var/scripts/apps/fulltextsearch.sh
But I have actually currently no idea :/

**Comment by enoch85** [comment]
@BetaLeaf Try to run this instead: https://raw.githubusercontent.com/nextcloud/vm/master/static/apps.sh

**Comment by BetaLeaf** [comment]
@enoch85 I tried a reinstall using the command you posted. Unfortunately, I get the same error.

```
PHP Fatal error:  Declaration of daita\MySmallPhpTools\Db\ExtendedQueryBuilder::exprLimitToDBFieldArray(string $field, array $values, bool $eq = true, bool $cs = true, string $alias = ''): OCP\DB\QueryBuilder\ICompositeExpression must be compatible with daita\MySmallPhpTools\IExtendedQueryBuilder::exprLimitToDBFieldArray(string $field, array $values, bool $cs = true, string $alias = ''): OCP\DB\QueryBuilder\ICompositeExpression in /var/www/nextcloud/apps/social/vendor/daita/my-small-php-tools/lib/Db/ExtendedQueryBuilder.php on line 49
Sorry but something went wrong. Please report this issue to https://github.com/nextcloud/vm/issues and include the output of the error message. Thank you!
sudo -u www-data php /var/www/nextcloud/occ fulltextsearch:index failed
```

**Comment by BetaLeaf** [comment]
> I just wonder why the path to the script is /var/scripts/apps/fulltextsearch.sh
> But I have actually currently no idea :/

If you see https://github.com/nextcloud/vm/blob/master/lib.sh line 9 and 871, this is the default location for the install scripts. 

Here is the part of the script where it ask you to install additional apps:
https://github.com/nextcloud/vm/blob/master/nextcloud-startup-script.sh#L425 which calls https://github.com/nextcloud/vm/blob/master/static/apps.sh, which then calls https://github.com/nextcloud/vm/blob/master/apps/additional_apps.sh

The function that installs the apps in additional_apps.sh is found at: https://github.com/nextcloud/vm/blob/master/lib.sh#L873

Hope this answers your question @szaimen

**Comment by Ark74** [comment]
@BetaLeaf check the app signature is correct, for some reason I tend to believe that you are getting a badly build or corrupt file, maybe some cache issue?.

**Comment by BetaLeaf** [comment]
@Ark74 Did I do this correctly?
```
root@cloud:~# nextcloud_occ integrity:check-app fulltextsearch
  - EXCEPTION:
    - class: OC\IntegrityCheck\Exceptions\InvalidSignatureException
    - message: Signature data not found.
root@cloud:~# nextcloud_occ integrity:check-app fulltextsearch_elasticsearch
  - EXCEPTION:
    - class: OC\IntegrityCheck\Exceptions\InvalidSignatureException
    - message: Signature data not found.
root@cloud:~# nextcloud_occ integrity:check-app files_fulltextsearch
  - EXCEPTION:
    - class: OC\IntegrityCheck\Exceptions\InvalidSignatureException
    - message: Signature data not found.
```

**Comment by szaimen** [comment]
> > I just wonder why the path to the script is /var/scripts/apps/fulltextsearch.sh
> > But I have actually currently no idea :/
> 
> If you see https://github.com/nextcloud/vm/blob/master/lib.sh line 9 and 871, this is the default location for the install scripts.
> 
> Here is the part of the script where it ask you to install additional apps:
> https://github.com/nextcloud/vm/blob/master/nextcloud-startup-script.sh#L425 which calls https://github.com/nextcloud/vm/blob/master/static/apps.sh, which then calls https://github.com/nextcloud/vm/blob/master/apps/additional_apps.sh
> 
> The function that installs the apps in additional_apps.sh is found at: https://github.com/nextcloud/vm/blob/master/lib.sh#L873
> 
> Hope this answers your question @szaimen

Hm no, if you have read correctly it downloads the script to the /var/scripts directory and not /var/scripts/apps. But I knkw that on earlier builds the app scripts where downloaded to /var/scripts/apps at during the install script. That is fhe reason why enoch85 suggested you to run this script https://raw.githubusercontent.com/nextcloud/vm/master/static/apps.sh which should download the newest fulltextsearch script...



> @Ark74 Did I do this correctly?
> 
> ```
> root@cloud:~# nextcloud_occ integrity:check-app fulltextsearch
>   - EXCEPTION:
>     - class: OC\IntegrityCheck\Exceptions\InvalidSignatureException
>     - message: Signature data not found.
> root@cloud:~# nextcloud_occ integrity:check-app fulltextsearch_elasticsearch
>   - EXCEPTION:
>     - class: OC\IntegrityCheck\Exceptions\InvalidSignatureException
>     - message: Signature data not found.
> root@cloud:~# nextcloud_occ integrity:check-app files_fulltextsearch
>   - EXCEPTION:
>     - class: OC\IntegrityCheck\Exceptions\InvalidSignatureException
>     - message: Signature data not found.
> ```

But this seems to be the problem...

**Comment by Ark74** [comment]
> But this seems to be the problem...

Well not really, I mean I just realize that the app is not signed.

But still, IIRC the error shown is from the apps used on the build process, maybe @daita knows on more detail.

@BetaLeaf could you enable debug on the script and share the output?

**Comment by BetaLeaf** [comment]
Yea, let me look up how to do that lol

**Comment by Ark74** [comment]
Check here please,
https://github.com/nextcloud/vm/blob/master/apps/fulltextsearch.sh#L16

**Comment by BetaLeaf** [comment]
Yea I'm just not sure how to set that, but I saw enoch85 post how once in an issue so I'm searching for that.

**Comment by BetaLeaf** [comment]
[debug.txt](https://github.com/nextcloud/vm/files/4273961/debug.txt)
@Ark74

**Comment by Ark74** [comment]
`root@cloud:~# nextcloud_occ fulltextsearch:test` ?

**Comment by BetaLeaf** [comment]
@Ark74 
```
root@cloud:/var/scripts/apps# nextcloud_occ fulltextsearch:test

.Testing your current setup:
Creating mocked content provider. ok
Testing mocked provider: get indexable documents. (2 items) ok
Loading search platform. (Elasticsearch) ok
Testing search platform. ok
Locking process ok
Removing test. ok
Pausing 3 seconds 1 2 3 ok
Initializing index mapping. ok
Indexing generated documents. ok
Pausing 3 seconds 1 2 3 ok
Retreiving content from a big index (license). (size: 32386) ok
Comparing document with source. ok
Searching basic keywords:
 - 'test' (result: 1, expected: ["simple"]) ok
 - 'document is a simple test' (result: 2, expected: ["simple","license"]) ok
 - '"document is a test"' (result: 0, expected: []) ok
 - '"document is a simple test"' (result: 1, expected: ["simple"]) ok
 - 'document is a simple -test' (result: 1, expected: ["license"]) ok
 - 'document is a simple +test' (result: 1, expected: ["simple"]) ok
 - '-document is a simple test' (result: 0, expected: []) ok
Updating documents access. ok
Pausing 3 seconds 1 2 3 ok
Searching with group access rights:
 - 'license' - [] -  (result: 0, expected: []) ok
 - 'license' - ["group_1"] -  (result: 1, expected: ["license"]) ok
 - 'license' - ["group_1","group_2"] -  (result: 1, expected: ["license"]) ok
 - 'license' - ["group_3","group_2"] -  (result: 1, expected: ["license"]) ok
 - 'license' - ["group_3"] -  (result: 0, expected: []) ok
Searching with share rights:
 - 'license' - notuser -  (result: 0, expected: []) ok
 - 'license' - user2 -  (result: 1, expected: ["license"]) ok
 - 'license' - user3 -  (result: 1, expected: ["license"]) ok
Removing test. ok
Unlocking process ok
```

**Comment by Ark74** [comment]
Well the (re)installation process works :wink: 

Seems to be a underlying issue on the app, maybe a regression on the last update, I guess @daita  can help us a little more on the matter.

Ok, I'll go get some sleep, I'll try to find a time to check this in more detail later in the week.
Regards!

**Comment by BetaLeaf** [comment]
Thanks for looking into this @Ark74.

**Comment by enoch85** [comment]
@Ark74 Thanks for helping out here! :rocket: 

@BetaLeaf Thanks for posting!

**Comment by enoch85** [comment]
I had the same issue yesterday, a quick Goolge search led me to [this](https://github.com/nextcloud/fulltextsearch/issues/373#issuecomment-418506328)

I truncated the table, and it started to work again.

Please try that.

**Comment by szaimen** [comment]
@enoch85 if that works, maybe we should think about adding this to the uninstall/reinstall process?

**Comment by enoch85** [comment]
> A if that works, maybe we should think about adding this to the uninstall/reinstall process?

Yeah I've thought about that as well. Since it doesn't do any harm according to @daita, maybe it would be a good idea.

**Comment by BetaLeaf** [comment]
@enoch85 Sorry for the delay. I have been sick. I redownloaded the latest (as of 5 minutes before this comment) fulltextsearch.sh from this repo and reinstalled it. Unfortunately, I am still having issues. The error code I am getting is as follows:
```
PHP Fatal error:  Declaration of daita\MySmallPhpTools\Db\ExtendedQueryBuilder::exprLimitToDBFieldArray(string $field, array $values, bool $eq = true, bool $cs = true, string $alias = ''): OCP\DB\QueryBuilder\ICompositeExpression must be compatible with daita\MySmallPhpTools\IExtendedQueryBuilder::exprLimitToDBFieldArray(string $field, array $values, bool $cs = true, string $alias = ''): OCP\DB\QueryBuilder\ICompositeExpression in /var/www/nextcloud/apps/social/vendor/daita/my-small-php-tools/lib/Db/ExtendedQueryBuilder.php on line 49
```

**Comment by BetaLeaf** [comment]
@enoch85 Issue fully resolved now. Seems the social app is causing issues with other apps and disabling social app and reinstalling fulltextsearch.sh fixes the issue.

https://github.com/nextcloud/social/issues/856

**Comment by enoch85** [comment]
OK, nice info! I will disable the social app in future versions.

**Comment by enoch85** [comment]
This will be fixed in 20.04.
